WifiART/src/userinterface/ArtMvcController.java

```diff
@@ -122,22 +124,118 @@ private void updateSubelementBuffer(SubelementName subelementName) {
 
     private void updateTotalBuffer() {
         try {
-            List<String> lciSubelementBuffersList = model.getLciSubelementBuffersList();
-            List<String> lcrSubelementBuffersList = model.getLcrSubelementBuffersList();
-            view.displayBuffer(lciSubelementBuffersList, lcrSubelementBuffersList, model.getState().isReadable());
-            writeBufferToFile(lciSubelementBuffersList, lcrSubelementBuffersList);
+            HashMap<SubelementName, String> lciSubelementBuffersList = model.getLciSubelementBuffersList();
+            HashMap<SubelementName, String> lcrSubelementBuffersList = model.getLcrSubelementBuffersList();
+            String readableBuffer = getReadableBufferDisplay(lciSubelementBuffersList, lcrSubelementBuffersList);
```

Comment:
nit: move this to the if condition below. i.e, only get readable buffer when the state is readable, and get nonreadable buffer when the state is nonreadable.

Reply:
Agreed. Fixed this in the revision.